overlord/state/taskrunner.go

```diff
@@ -73,13 +88,17 @@ func (r *TaskRunner) run(fn HandlerFunc, task *Task) {
 
 		r.state.Lock()
 		defer r.state.Unlock()
+		halted := task.HaltTasks()
 		if err == nil {
```

Comment:
I'd swap the logic (so test ``err != nil``) and return early but don't block on this if you disagree.